ros2_batch_job/windows_batch/__init__.py

```diff
@@ -45,10 +45,8 @@ def setup_env(self):
         with open('env.bat', 'w') as f:
             f.write("@echo off" + os.linesep)
             assert self.args.visual_studio_version is not None
-            f.write(
-                'call '
-                '"C:\\Program Files (x86)\\Microsoft Visual Studio\\%s\\Community\\VC\\Auxiliary\\Build\\vcvarsall.bat" ' %
-                self.args.visual_studio_version + 'x86_amd64' + os.linesep)
+            vs = self.args.visual_studio_version
+            f.write(f'call "C:\\Program Files (x86)\\Microsoft Visual Studio\\{vs}\\BuildTools\\VC\\Auxiliary\\Build\\vcvarsall.bat" x86_amd64' + os.linesep)
```

Comment:
I think this is a fair compromise and removes the problem of docker CMD execution string expansion that has weird and unexpected behavior.